salt/orch/update.sls

```diff
@@ -83,6 +83,33 @@ pre-orchestration-migration:
     - require:
       - update-modules
 
+# NOTE: Remove me for 4.0
+#
+# During an upgrade from 2.0 to 3.0, as we go master by master first, the first master will not
+# succeed on the orchestration if it doesn't have an etcd member. Assume M{1,2,3}, W{1,2}. Assume
+# etcd members are running on M2, W1 and W2.
+#
+# M1 updates its configurations on highstate and refers to the etcd nodes with the new names
+# (hostnames) instead of machine-ids, but M2, W1 and W2 still didn't run anything to refresh their
+# certificates and their etcd instances, what will make M1 fail because it cannot connect to any
+# etcd instance (as all certificates look invalid at this point for M2.hostname, W1.hostname and
+# W2.hostname). This makes the apiserver on M1 fail restarting itself until the orchestration reaches
+# M2 [there's no hard dependency on states between masters], but the orchestration already failed on
+# M1, so the global result will be failure nevertheless.
```

Comment:
Cound't we just update `/etc/hosts` in all the masters in the cluster prior to starting the update orchestration? The new `/etc/hosts` will contain both hostnames and machine-ids...

Reply:
It wouldn't help, as the certificates on the running etcd members would still not contain the new name in the SANs, what would still isolate M1 when upgrading (in the presented case in the comment).

Reply:
I see.